pkg/builders/buildpacks/builder_test.go

```diff
@@ -116,6 +118,45 @@ func TestBuild_BuilderImageConfigurable(t *testing.T) {
 	}
 }
 
+// TestBuild_BuilderImageExclude ensures that ignored files are not added to the func
+// image
+func TestBuild_BuilderImageExclude(t *testing.T) {
+	var (
+		i = &mockImpl{} // mock underlying implementation
+		b = NewBuilder( // Func Builder logic
+			WithName(builders.Pack), WithImpl(i))
+		f = fn.Function{
+			Runtime: "go",
+		}
+	)
+	funcIgnoreContent := []byte(`#testing comments
+hello.txt`)
+	expected := []string{"hello.txt"}
+
+	tempdir := t.TempDir()
+	f.Root = tempdir
+
+	//create a .funcignore file containing the details of the files to be ignored
+	err := os.WriteFile(filepath.Join(f.Root, ".funcignore"), funcIgnoreContent, 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	i.BuildFn = func(ctx context.Context, opts pack.BuildOptions) error {
+		if len(opts.ProjectDescriptor.Build.Exclude) != 2 {
```

Comment:
Wait, why is this 2 not 1? The comment should be ignored, right?

Reply:
btw good way to diff two arrays in test is `cmp.Diff()`

Reply:
> Wait, why is this 2 not 1? The comment should be ignored, right?

because we are not excluding the comment lines ourselves, we are passing all the lines directly to the exclude option of buildpack which is doing this using https://github.com/sabhiram/go-gitignore/blob/master/ignore.go. So, it will a duplication of code if we does that here as well.

Reply:
> btw good way to diff two arrays in test is `cmp.Diff()`

thanks for the info, will keep that in mind while comparing arrays.